workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfigurationEditor.cs

```diff
@@ -10,60 +10,79 @@ namespace Improbable.Gdk.BuildSystem.Configuration
+            using (new EditorGUILayout.HorizontalScope())
             {
-                if (workerConfiguration.WorkerBuildConfigurations.All(x => x.WorkerType != workerTypeName))
+                workerTypeName = EditorGUILayout.TextField(workerTypeName);
+
+                GUILayout.FlexibleSpace();
+
+                var canAddWorker = string.IsNullOrEmpty(workerTypeName) ||
+                    workerConfiguration.WorkerBuildConfigurations.Any(c =>
+                        c.WorkerType.Equals(workerTypeName, StringComparison.InvariantCultureIgnoreCase));
+
+                using (new EditorGUI.DisabledScope(canAddWorker))
                 {
-                    var config = new WorkerBuildConfiguration
+                    if (GUILayout.Button(AddWorkerTypeButtonContents, EditorStyles.miniButton))
                     {
-                        WorkerType = workerTypeName,
-                        LocalBuildConfig = new BuildEnvironmentConfig
-                        {
-                            BuildOptions = BuildOptions.Development,
-                        },
-                        CloudBuildConfig = new BuildEnvironmentConfig
+                        EditorUtility.SetDirty(target);
+                        Undo.RecordObject(target, $"Remove '{workerTypeName}'");
+
+                        var config = new WorkerBuildConfiguration
                         {
-                            BuildOptions = BuildOptions.EnableHeadlessMode,
-                        }
-                    };
-                    workerConfiguration.WorkerBuildConfigurations.Add(config);
+                            WorkerType = workerTypeName,
+                            LocalBuildConfig = new BuildEnvironmentConfig
+                            {
+                                BuildOptions = BuildOptions.Development,
+                            },
+                            CloudBuildConfig = new BuildEnvironmentConfig
+                            {
+                                BuildOptions = BuildOptions.EnableHeadlessMode,
+                            }
+                        };
+                        workerConfiguration.WorkerBuildConfigurations.Add(config);
+                    }
                 }
             }
 
-            EditorGUILayout.EndHorizontal();
+            var rect = EditorGUILayout.GetControlRect(false, 1, EditorStyles.foldout);
+            var oldColor = Handles.color;
+            Handles.color = new Color(0.3f, 0.3f, 0.3f, 1);
+
+            Handles.DrawLine(new Vector2(rect.x, rect.yMax), new Vector2(rect.xMax, rect.yMax));
+            Handles.color = oldColor;
+
+            GUILayout.Space(rect.height);
 
             var configs = workerConfiguration.WorkerBuildConfigurations;
             foreach (var workerConfig in configs)
             {
                 if (!DrawWorkerConfiguration(workerConfig))
                 {
+                    EditorUtility.SetDirty(target);
```

Comment:
`Undo.RecordObject` already records the object as dirty according to https://docs.unity3d.com/ScriptReference/EditorUtility.SetDirty.html

Reply:
Original code had it, I'll test and remove them all if it works as expected